--- a/configs/linux-5.15/sun55iw3p1.dtsi
+++ b/configs/linux-5.15/sun55iw3p1.dtsi
@@ -4960,12 +4960,6 @@
 			clocks = <&ccu CLK_GPU>,
 				<&ccu CLK_BUS_GPU>,
-				<&ccu CLK_PLL_GPU>,
-				<&ccu CLK_PLL_PERI0_800M>,
-				<&ccu CLK_PLL_PERI0_600M>,
-				<&ccu CLK_PLL_PERI0_400M>,
-				<&ccu CLK_PLL_PERI0_300M>,
-				<&ccu CLK_PLL_PERI0_200M>;
-			clock-names = "clk_mali", "clk_bus", "clk_parent",
-				"clk_800", "clk_600", "clk_400", "clk_300", "clk_200";
+				<&ccu CLK_PLL_GPU>;
+			clock-names = "clk_mali", "clk_bus", "clk_parent";
 			resets = <&ccu RST_BUS_GPU>;
 			reset-names = "rst_bus_gpu";
@@ -4987,5 +4981,6 @@
 		gpu_opp_table: gpu-opp-table {
 			compatible = "allwinner, mali-valhall-operating-points";
-			/*opp@150000000 {
+
+			opp@150000000 {
 				opp-hz = /bits/ 64 <150000000>;
 				opp-microvolt = <900000>;
@@ -5000,5 +4995,5 @@
 				opp-hz = /bits/ 64 <300000000>;
 				opp-microvolt = <900000>;
-			};*/
+			};
 
 			opp@400000000 {
@@ -5014,43 +5009,27 @@
 			opp@648000000 {
 				opp-hz = /bits/ 64 <648000000>;
-				opp-microvolt = <0>;
-				opp-microvolt-vf0900 = <900000>;
+				opp-microvolt = <900000>;
 			};
 
 			opp@696000000 {
 				opp-hz = /bits/ 64 <696000000>;
-				opp-microvolt = <0>;
-				opp-microvolt-vf1920 = <900000>;
-				opp-microvolt-vf2920 = <900000>;
-				opp-microvolt-vf3920 = <900000>;
-				opp-microvolt-vf21920 = <900000>;
-				opp-microvolt-vf31920 = <900000>;
-				opp-microvolt-vf5920 = <900000>;
+				opp-microvolt = <920000>;
 			};
 
 			opp@744000000 {
 				opp-hz = /bits/ 64 <744000000>;
-				opp-microvolt = <0>;
-				opp-microvolt-vf4920 = <900000>;
-				/* Not use: only for performance test
-				   opp-microvolt-vf2920 = <900000>;
-				   opp-microvolt-vf3920 = <900000>;
-				   opp-microvolt-vf21920 = <900000>;
-				   opp-microvolt-vf31920 = <900000>;
-				*/
-			};
-
-			/* Not use: only for performance test */
-			opp@792000000 {
-				opp-hz = /bits/ 64 <792000000>;
-				opp-microvolt = <0>;
-				/* opp-microvolt-vf2950 = <900000>;
-				   opp-microvolt-vf3950 = <900000>;
-				   opp-microvolt-vf21950 = <900000>;
-				   opp-microvolt-vf31950 = <900000>;
-				*/
-			};
-		};
-
+				opp-microvolt = <1000000>;
+			};
+
+			opp@840000000 {
+				opp-hz = /bits/ 64 <840000000>;
+				opp-microvolt = <1000000>;
+			};
+
+			opp@888000000 {
+				opp-hz = /bits/ 64 <888000000>;
+				opp-microvolt = <1050000>;
+			};
+		};
 		combophy: phy@4f00000 {
 			compatible = "allwinner,inno-combphy";
