--- a/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_platform.c
+++ b/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_platform.c
@@ -1226,7 +1226,7 @@
 		p--;
 
 	p++;
-	strncpy(val, dst, p -dst);
+	memcpy(val, dst, p - dst);
 	val[p - dst] = 0;
 	return 0;
 }
--- a/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_msg_tx.c
+++ b/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_msg_tx.c
@@ -4205,7 +4205,7 @@
 		return -ENOMEM;
 
 	/* Set parameters for the MM_DBG_TRIGGER_REQ message */
-	strncpy(req->error, msg, sizeof(req->error));
+	strscpy(req->error, msg, sizeof(req->error));
 
 	/* Send the MM_DBG_TRIGGER_REQ message to LMAC FW */
 	return rwnx_send_msg(rwnx_hw, req, 0, -1, NULL);
--- a/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_main.c
+++ b/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_main.c
@@ -3955,7 +3955,8 @@
 static int
 rwnx_cfg80211_remain_on_channel(struct wiphy *wiphy, struct wireless_dev *wdev,
 								struct ieee80211_channel *chan,
-								unsigned int duration, u64 *cookie)
+								unsigned int duration, u64 *cookie,
+								const u8 *rx_addr)
 {
 	return rwnx_cfg80211_remain_on_channel_(wiphy, wdev, chan, duration, cookie, false);
 }
